src/polyhedralGravity/calculation/SanityCheck.cpp

```diff
@@ -0,0 +1,76 @@
+#include "SanityCheck.h"
+
+namespace polyhedralGravity::SanityCheck {
+
+    bool checkNormalsOutwardPointing(const Polyhedron &polyhedron) {
+        using namespace util;
+        auto it = GravityModel::transformPolyhedron(polyhedron);
+        // All normals have to point outwards (intersect the polyhedron even times)
```

Comment:
not sure if you check this elsewhere, but if your triangle is in some way degenerate (two points the same, all three collinear) this (and other code) could break, I think? Maybe we should check that too since it's linear?

Reply:
Good idea, I added this check 👍